From b42dd9c69f41a105f0cfe4b1e64e1d915f03dade Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 18 Oct 2017 13:09:27 +0000
Subject: [PATCH 44/44] macOS: Workaround Cocoa/Boost 'check' macro conflict

Apple have a macro in Cocoa somewhere called check()
which conflicts with the Boost assert macros.

See: https://github.com/facebook/osquery/pull/3613/files
and: https://svn.boost.org/trac10/ticket/6219
---
 src/cpp/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index e05a1ba75b..7cf6dd74c5 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -94,6 +94,11 @@ if(UNIX)
             set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} -stdlib=libstdc++")
          endif()
       endif()
+      # Apple have a macro in Cocoa somewhere called check()
+      # which conflicts with the Boost assert macros.
+      # See: https://github.com/facebook/osquery/pull/3613/files
+      # and: https://svn.boost.org/trac10/ticket/6219
+      add_definitions(-D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0)
    endif()
 
    # gcc hardending options (see: http://wiki.debian.org/Hardening)
-- 
2.14.1

